lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -435,17 +435,17 @@ def add_splits(post, options)
 
         splits = []
         split_data.each do |split|
-          amount = {
-            value: split['amount']['value']
-          }
-          amount[:currency] = split['amount']['currency'] if split['amount']['currency']
-
-          split_hash = {
-            amount: amount,
-            type: split['type'],
-            reference: split['reference']
-          }
-          split_hash['account'] = split['account'] if split['account']
+          if split['amount']
+            amount = {}
+            amount[:value] = split['amount']['value'] if split['amount']['value']
+            amount[:currency] = split['amount']['currency'] if split['amount']['currency']
+          end
+
+          split_hash = {}
+          split_hash[:amount] = amount unless split['type'] == 'PaymentFee' || split['type'] == 'AdyenFees' || amount.nil?
+          split_hash[:type] = split['type'] if split['type']
+          split_hash[:reference] = split['reference'] if split['reference']
+          split_hash[:account] = split['account'] if split['account']
```

Comment:
I notice this is the one place where you've changed the syntax from string to symbol 🧐

Reply:
hmm now that you mentioned it, i wonder why the split_hash['account'] wasnt a symbol to begin with but the other split_hash fields are?

Reply:
Should I leave it as it is now (it'd be kinda weird) or just change it to symbol to keep these fields uniformed?

Reply:
For posterity - we chatted about this and decided to leave as is since it's currently working